### PR TITLE
Use FMV for transfers

### PR DESCRIPTION
**Pull request by gabmontes**
### Description

The subgraph valued incoming sVUSD transfers at 0 cost basis. When users round-tripped shares through DeFi protocols (Curve LPs, routers, etc.), cost basis "leaked" on every cycle — the outgoing transfer removed cost proportionally, but the returning shares added nothing back. This produced phantom earnings.

#### Fix

Value incoming transfers at fair market value using `convertToAssets(shares)` at the block of receipt (falls back to 1:1 if the call reverts). This matches how Aave's scaled-balance model implicitly handles transferred aTokens — the industry-standard approach for transferable yield-bearing tokens.

The from-side proportional reduction is unchanged (correct under average cost accounting).

#### Auto summary

This pull request updates the handling of share transfers in the staking subgraph to track cost basis more accurately. Instead of keeping the receiving user's `totalCostBasis` unchanged on transfer-in, the cost basis is now incremented by the fair market value (FMV) of the incoming shares, ensuring that metrics remain correct even when shares are routed through DeFi protocols or the vault's exchange rate changes. The code is also made more robust to contract upgrades and edge cases. The test suite is updated accordingly to validate these new behaviors.

**Core logic changes:**

* The `handleTransfer` function now increments the receiving user's `totalCostBasis` by the FMV of incoming shares, using `convertToAssets` to determine value, and falls back to a 1:1 rate if the call reverts (e.g., empty vault edge case). [[1]](diffhunk://#diff-21fb7fcabbd216361917473329a8f2b8f8204e46e9cc18ff473fdca61ca48e9eL234-R248) [[2]](diffhunk://#diff-21fb7fcabbd216361917473329a8f2b8f8204e46e9cc18ff473fdca61ca48e9eR260)
* Defensive coding: The transfer handler now gracefully exits if the sender's staking position does not exist, to handle potential future contract upgrades.

**Testing improvements:**

* The test suite for `handleTransfer` is updated to check that cost basis is set at FMV for transfer-ins, not left unchanged or set to zero. Tests now mock `convertToAssets` at a 1.2:1 rate and verify correct cost basis calculation for both new and existing users. [[1]](diffhunk://#diff-b74dd3f95bad2b490bb56c7f921b2d74b20de91c8bbd17d066732522d2f892a2R827-R847) [[2]](diffhunk://#diff-b74dd3f95bad2b490bb56c7f921b2d74b20de91c8bbd17d066732522d2f892a2L854-R872) [[3]](diffhunk://#diff-b74dd3f95bad2b490bb56c7f921b2d74b20de91c8bbd17d066732522d2f892a2L875-R892) [[4]](diffhunk://#diff-b74dd3f95bad2b490bb56c7f921b2d74b20de91c8bbd17d066732522d2f892a2L900-R921) [[5]](diffhunk://#diff-b74dd3f95bad2b490bb56c7f921b2d74b20de91c8bbd17d066732522d2f892a2L909-R930) [[6]](diffhunk://#diff-b74dd3f95bad2b490bb56c7f921b2d74b20de91c8bbd17d066732522d2f892a2L929-R954) [[7]](diffhunk://#diff-b74dd3f95bad2b490bb56c7f921b2d74b20de91c8bbd17d066732522d2f892a2L944-R968) [[8]](diffhunk://#diff-b74dd3f95bad2b490bb56c7f921b2d74b20de91c8bbd17d066732522d2f892a2L961-R990) [[9]](diffhunk://#diff-b74dd3f95bad2b490bb56c7f921b2d74b20de91c8bbd17d066732522d2f892a2R1005-R1007) [[10]](diffhunk://#diff-b74dd3f95bad2b490bb56c7f921b2d74b20de91c8bbd17d066732522d2f892a2L1000-R1032) [[11]](diffhunk://#diff-b74dd3f95bad2b490bb56c7f921b2d74b20de91c8bbd17d066732522d2f892a2L1012-R1044) [[12]](diffhunk://#diff-b74dd3f95bad2b490bb56c7f921b2d74b20de91c8bbd17d066732522d2f892a2R1059-R1061) [[13]](diffhunk://#diff-b74dd3f95bad2b490bb56c7f921b2d74b20de91c8bbd17d066732522d2f892a2R1080-R1081)
* Added a new test to verify fallback behavior when `convertToAssets` reverts, ensuring cost basis defaults to a 1:1 rate.

### Checklist

- [ ] Manual testing passed.
- [x] Automated tests added, or N/A.
- [x] Documentation updated, or N/A.
- [x] Environment variables set in CI, or N/A.

### CODE REVIEW

**Review by gndelia — DISMISSED** [review]

File: subgraph/src/earn.ts [hunk 1]
```diff
@@ -207,7 +207,13 @@ export function handleTransfer(event: Transfer): void {
...
   // Update the sending party: decrement shares and totalCostBasis proportionally.
   const fromId = buildId(vaultAddress, event.params.from.toHexString());
-  const fromEntity = UserStakingPosition.load(fromId)!;
+  // Defensive: the entity should always exist (created by handleDeposit or a
+  // prior transfer-in), but the contract is upgradeable — bail out gracefully
+  // if a future code path mints shares without emitting Deposit.
+  const fromEntity = UserStakingPosition.load(fromId);
+  if (fromEntity == null) {
+    return;
+  }
```
[the remark this answers is not in the document]
Comment (gabmontes) [reply subgraph/src/earn.ts]: Not an issue.

File: subgraph/src/earn.ts [hunk 2]
```diff
@@ -207,7 +207,13 @@ export function handleTransfer(event: Transfer): void {
 
   // Update the sending party: decrement shares and totalCostBasis proportionally.
   const fromId = buildId(vaultAddress, event.params.from.toHexString());
-  const fromEntity = UserStakingPosition.load(fromId)!;
+  // Defensive: the entity should always exist (created by handleDeposit or a
+  // prior transfer-in), but the contract is upgradeable — bail out gracefully
+  // if a future code path mints shares without emitting Deposit.
```
[the remark this answers is not in the document]
Comment (gabmontes) [reply subgraph/src/earn.ts]: Check removed.

**Review by gndelia — APPROVED** [review]

**Review by ArturDolzan — APPROVED** [review]
